examples/asr/librispeech_ctc_decoder/inference.py

```diff
@@ -0,0 +1,119 @@
+import argparse
+from typing import Optional
+
+import torch
+import torchaudio
+from torchaudio.prototype.ctc_decoder import kenlm_lexicon_decoder
+
+
+def _download_files(lexicon_file, kenlm_file):
+    torch.hub.download_url_to_file(
+        "https://pytorch.s3.amazonaws.com/torchaudio/tutorial-assets/ctc-decoding/lexicon-librispeech.txt", lexicon_file
+    )
+    torch.hub.download_url_to_file(
+        "https://pytorch.s3.amazonaws.com/torchaudio/tutorial-assets/ctc-decoding/4-gram-librispeech.bin", kenlm_file
+    )
+
+
+def run_inference(args):
+    # get pretrained wav2vec2.0 model
+    bundle = getattr(torchaudio.pipelines, args.model)
+    model = bundle.get_model()
+    tokens = [label.lower() for label in bundle.get_labels()]
+
+    # get decoder files
+    hub_dir = torch.hub.get_dir()
+    lexicon_file = f"{hub_dir}/lexicon.txt"
+    kenlm_file = f"{hub_dir}/kenlm.bin"
+    _download_files(lexicon_file, kenlm_file)
+
+    decoder = kenlm_lexicon_decoder(
+        lexicon=lexicon_file,
+        tokens=tokens,
+        kenlm=kenlm_file,
+        nbest=1,
+        beam_size=1500,
+        beam_size_token=None,
+        beam_threshold=50,
+        lm_weight=args.lm_weight,
+        word_score=args.word_score,
+        unk_score=float("-inf"),
+        sil_score=0,
+        log_add=False,
+    )
+
+    dataset = torchaudio.datasets.LIBRISPEECH(args.librispeech_path, url=args.split, download=False)
+
+    total_edit_distance = 0
+    total_length = 0
+    for idx, sample in enumerate(dataset):
+        waveform, _, transcript, _, _, _ = sample
+        transcript = transcript.strip().lower().strip()
+
+        with torch.inference_mode():
+            emission, _ = model(waveform)
+        results = decoder(emission)
+        hypothesis = " ".join(results[0][0].words).lower().strip()
+
+        total_edit_distance += torchaudio.functional.edit_distance(transcript.split(), hypothesis.split())
+        total_length += len(transcript.split())
+
+        if idx % 100 == 0:
+            print(f"Processed elem {idx}; WER: {total_edit_distance / total_length}")
```

Comment:
Can we use `logging` module and add timestamp? So that users can have some idea on how long it will take, or look back the log and see how long it took.